transportable-udfs-spark_2.11/src/main/scala/com/linkedin/transport/spark/StdUdfWrapper.scala

```diff
@@ -47,7 +47,10 @@ abstract class StdUdfWrapper(_expressions: Seq[Expression]) extends Expression
     _stdUdf = _sparkTypeInference.getStdUdf
     _nullableArguments = _stdUdf.getAndCheckNullableArguments
     _stdUdf.init(_stdFactory)
-    getRequiredFiles()
+    // Only call this on driver
+    if (TaskContext.get == null) {
+      getRequiredFiles()
+    }
```

Comment:
Since we only call `getRequiredFiles()` on the driver, `_distributedCacheFiles` will be `null` on executor nodes.

But within `eval()` (on executors), we call `processRequiredFiles()`:
https://github.com/linkedin/transport/blob/afcfea18af5d92243be861c608463e0a1e654b66/transportable-udfs-spark_2.11/src/main/scala/com/linkedin/transport/spark/StdUdfWrapper.scala#L139-L141

This in turn will attempt to access `_distributedCacheFiles`:
https://github.com/linkedin/transport/blob/afcfea18af5d92243be861c608463e0a1e654b66/transportable-udfs-spark_2.11/src/main/scala/com/linkedin/transport/spark/StdUdfWrapper.scala#L191-L193

Won't this cause an NPE?

Reply:
I think the current code works based on the following assumption: since `_distributedCacheFiles` is not a transient field, once it's initialized correctly on the driver, it will be serialized and sent to the executor, and the reconstructed object on the executor will already have the `_distributedCacheFiles` initialized.

Reply:
Ah, right ... Good catch, thanks! I was wondering how this was possibly working as-is.